### PR TITLE
Improve MMU mask checks

### PR DESCRIPTION
**Pull request by blubberdiblub**
The MMU code uses some inefficient and not very readable validity check for masks. So I went ahead and created an efficient function (it's the counterpart of https://graphics.stanford.edu/~seander/bithacks.html#DetermineIfPowerOf2) and replaced the 2 instances with calls to it.

The first and unreadable instance that used a loop(!) was in the SDR register update code and the second (better, but not optimal) instance that used the popcount function was in the BAT register update code.

I suspect that there will be no measurable performance gains as I guess the updates happen relatively seldom and the SDR update should only be relevant for those rare games that make use of page tables. However, there should be significant readability improvements.

### CODE REVIEW

**Comment by RisingFog** [comment]
@dolphin-emu-bot rebuild

**Comment by blubberdiblub** [comment]
BTW, I ran Star Wars - Rogue Squadron II - Rogue Leader (GSWP64) for testing, which worked.

**Comment by Tilka** [comment]
LGTM.

File: Source/Core/Common/BitUtils.h [hunk 1]
```diff
@@ -99,4 +99,26 @@ constexpr Result ExtractBits(const T src) noexcept
 
   return ExtractBits<T, Result>(src, begin, end);
 }
+
+///
+/// Verifies whether the supplied value is a valid bit mask of the form 0b00...0011...11.
+/// Both edge cases of all zeros and all ones are considered valid masks, too.
+///
+/// @param  mask The mask value to test for validity.
```
Comment (shuffle2) [inline Source/Core/Common/BitUtils.h]: not really a fan of this comment style (which isn't used anywhere else in dolphin?), but w/e

Comment (blubberdiblub) [inline Source/Core/Common/BitUtils.h]: Me neither, I was just copying the same style that was already used in the file.
If you point me to a file that demonstrates the recommended style, I can change the whole file, if desired.

Comment (shuffle2) [inline Source/Core/Common/BitUtils.h]: oh, I guess @lioncash is to blame for this cancer then :P

Comment (lioncash) [inline Source/Core/Common/BitUtils.h]: Yeah, the original code came from a utility file from an [ARMv7](https://bitbucket.org/Lioncash/forearm/src/963a94b5011e2f5a2d0e874d63118430c8a3e888/libForeARM/src/Utils/Bits.hpp?at=master&fileviewer=file-view-default) interpreter that I'm working on.
